tests/Unittest-README.md

```diff
@@ -0,0 +1,159 @@
+# Ansible CVP Unit testing
+
+## Unit test structure
+The structure of unittests are described below
+- **`tests/unit`**: Contains all unittests 
+  - Filename start with `test_` followed by the name of the file it's testing. 
+    - For eg: Unittests for `ansible_collection/arista/cvp/plugins/module_utils/device_tools.py` will be under `tests/unit/test_cv_device_tools.py`
+  - `tests/unit/conftest.py`: This is a special python script that defines all the fixtures available for use while testing `ansible-cvp` modules
+- **`tests/lib`**: Contains library functions
+  - **`tests/lib/mockMagic.py`**: Contains mock cvprac functions
+    - Note: the name and signature of the mock functions matches original cvprac functions
+- **`tests/data`**: Contains faked datasets for use by the mocked cvprac functions
+
+## Write unittests
+### General guidelines
+- Create your unittest file under `tests/unit` following the naming convention mentioned in the [above section](#unit-test-structure)
+- If the file is already present, add your unittests in the file. 
+- Each function in the __file-under-test__(eg: `ansible_collection/arista/cvp/plugins/module_utils/device_tools.py`) should be placed in a class of it's own with the class name starting with `Test` followed by the name of the function.
+
+```python
+class TestValidateConfig():
+    """
+    Contains unit tests for validate_config()
+    """
+...
+```
+- Add tags to run cases selectively
+
+```python
+@pytest.mark.api
+@pytest.mark.configlet
+```
+
+> Tags must be defined in [pytest.ini](./pytest.ini) file.
+
+> Tags can be added at `class` or `test` level.
+```python
+@pytest.mark.api
+class TestFoo(..):
+  ...
+
+class TestBar(...):
+  @pytest.mark.configlet
+  def test_bar_scenario1(...):
+    ...
+```
+- Place the test suite within this class.
+
+```python
+class TestValidateConfig():
+    """
+    Contains unit tests for validate_config()
+    """
+    def test_validate_config_warning(...):
+      pass
+    def test_validate_config_error(...):
+      pass
+    ....
+```
+### Fixtures
+Fixtures are used to prepare the context for the tests. It can include environment(_`CVP` configured with some topology_) as well as content(_`CVP` configured with some configlet_). Some important fixures are described below. For more information, please refer to this file: `tests/unit/conftest.py`
```

Comment:
Add spacing between (), I would change wording as well. Instead of "please refer to this file: `test/unit/conftest.py`." Change that to something like please refer to the [conftest](tests/unit/conftest,py or whatever the real path is) file.